themes/default/content/blog/examining-recommended-practices-code-placement-and-stacks/index.md

```diff
@@ -0,0 +1,148 @@
+The discussion above is working from the assumption that Pulumi users will store their Pulumi code in a version control system, like Git. Strictly speaking, using a version control system isn't required for Pulumi to function (there is one exception we'll touch on later in this series when we discuss [Pulumi Deployments](https://www.pulumi.com/docs/intro/pulumi-service/deployments/)), but we **very strongly** recommend that all users use Pulumi in conjunction with version control.
+{{% /notes %}}
+
+In the case of Zephyr, their team felt like a monorepo approach (storing Pulumi code and application code together) was most appropriate for their specific requirements. Zephyr is a relatively small organization with a single team of developers that are responsible for managing both the cloud infrastructure as well as the development and deployment of the online store application. Zephyr's team also decicded to use a single Pulumi project---for now. (Be sure to follow this blog series to see how that evolves, and why!)
+
+### Addressing the need for multiple environments
+
+The second question Zephyr encountered is how to handle the need for multiple instances of their application. In addition to a production instance---which is the instance behind Zephyr's public-facing online store---Zephyr also felt they needed an environment for the developers to use in testing changes to the online store.
+
+This use case---needing to have multiple, separate instances of the infrastructure and applications created by a single Pulumi program---is exactly what [Pulumi stacks](https://www.pulumi.com/docs/intro/concepts/stack/) were designed to address. Each stack is a separate instance of the resources created by a Pulumi program within a project. Further, each stack has its own independent state, and each stack has its own configuration values. Stacks can be short-lived (meaning the associated resources are also short-lived), or stacks can be long-lived. Aside from the cloud resources created the Pulumi program, stacks are lightweight and simple to create or delete.
+
+<!--Are there other considerations that are involved in deciding whether or not to use stacks?-->
+
+Zephyr decided to initially start with two stacks: a production stack (named "prod") and a development stack (named "dev"). As you'll observe throughout this series, this is a decision that is easily adjusted over time as your organization's requirements change.
+
+## Viewing the first iteration of Zephyr's code
+
+You can view the first iteration of Zephyr's Pulumi and application code---the iteration that corresponds to the decisions described in this blog post---by navigating to [this GitHub repository](https://github.com/pulumi/zephyr-app/). From the branch/tag dropdown, switch from the "main" branch to viewing the "original" tag.
+
+From that GitHub repository, you can also choose to deploy the Pulumi code yourself. <!--Add instructions after repo tweaks are finished-->
```

Comment:
What do you think about adding a simple `template` to the config and Deploy with Pulumi button? I'd be happy to wire this up for you if you're okay with it.

Reply:
I'm fine with this, but I don't want to hold up the blog post. So, assuming we can add this later (seems like we should be able to), then let's do it. In the post, I'll just reference instructions in the repo (that way we can update the repo when we add the template).

Reply:
Oh absolutely, I was thinking completely separate from this post. More referring to the content of the repo, which can be updated later and out of band.

Ok, I'll add that to my post-week to-do list, then.